docs/spec/spec.md

```diff
@@ -2758,9 +2763,215 @@ public function main() returns error? {
+
+###### Example: Federated Subgraph
+
+```ballerina
+@subgraph:Subgraph
+service on new graphql:Listener(9090) {
+    // ...
+}
+```
+
+#### 13.1.2 The `@subgraph:Entity` Annotation
+
+In a federated graph, an entity is an object type that can resolve its fields across multiple subgraphs. Each subgraph can contribute different fields to the entity and is responsible for resolving only the fields that it contributes. The `@subgraph:Entity` designates an object type as an entity in Ballerina. The following type definition describes the shape of the `@subgraph:Entity` annotation.
+
+```ballerina
+# Describes the shape of the `subgraph:Entity` annotation
+# + key - GraphQL fields and subfields that contribute to the entity's primary key/keys
+# + resolveReference - Function pointer to resolve the entity
+public type FederatedEntity record {|
+    string|string[] key;
+    ReferenceResolver? resolveReference;
+|};
+
+# The annotation to designate a GraphQL object type as a federated entity.
+public annotation FederatedEntity Entity on class, type;
+```
+
+To fully define an entity within a Ballerina GraphQL subgraph, you must:
+
+1. Assign the `@subgraph:Entity` annotation to an object type.
+2. Define the `key` field of the annotation to be the fields and subfields that contribute to the entity's primary key/keys.
+3. Define the `resolveReference` field of the annotation to be a function pointer to resolve the entity. If this field is set to `nil`, it indicates to the graph router that this subgraph does not define a reference resolver for this entity. For more details, see [ReferenceResolver](#133-the-referenceresolver).
+
+###### Example: Federated Entity Definition and Corresponding GraphQL Schema
+
+<table>
+    <tr>
+        <th>Example</th>
+        <th>Ballerina definition</th>
+        <th>GraphQL schema</th>
+    </tr>
+    <tr>
+        <td>
+            Simple key
+        </td>
+        <td>
+            <pre lang="ballerina">
+```ballerina
+@subgraph:Entity {
+    key: "id",
+    resolveReference: resolveProduct
+}
+type Product record {
+    string id;
+    string name;
+    int price;
+};
+```
+            </pre>
+        </td>
+        <td>
+            <pre lang="graphql">
+```graphql
+type Product @key(fields: "id") {
+    id: String!
+    name: String!
+    price: Int!
+}
+```
+            </pre>
+        </td>
+    </tr>
+    <tr>
+        <td>
+            Mutiple keys
+        </td>
+        <td>
+            <pre lang="ballerina">
+```ballerina
+@subgraph:Entity {
+    key: ["id", "sku"],
+    resolveReference: resolveProduct
+}
+type Product record {
+    string id;
+    string name;
+    int price;
+};
+```
+            </pre>
+        </td>
+        <td>
+            <pre lang="graphql">
+```graphql
+type Product @key(fields: "id") @key(fields: "sku") {
```

Comment:
What is `sku` here?

Reply:
`sku` should be one of the fields of Product type. Updated.